examples/api-wrapper/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+TOKEN - SmartThings API token
+*/
+
+use std::env;
+use fastedge::{
+    body::Body,
+    http::{header, Error, Method, Request, Response, StatusCode},
+};
+use url::Url;
+
+const API_BASE: &str = "https://api.smartthings.com/v1/devices/";
+
+#[fastedge::http]
+fn main(req: Request<Body>) -> Result<Response<Body>, Error> {
+    match req.method() {
+        &Method::GET | &Method::HEAD => (),
+        _ => return Response::builder().status(StatusCode::METHOD_NOT_ALLOWED).header(header::ALLOW, "GET, HEAD").body(Body::from("This method is not allowed\n"))
+    };
+
+    let expected_pass = match env::var("PASSWORD") {
+        Err(_) => return Response::builder().status(StatusCode::INTERNAL_SERVER_ERROR).body(Body::from("Misconfigured app\n")),
+        Ok(r) => r
+    };
+    let provided_pass = match req.headers().get(header::AUTHORIZATION) {
+        None => return Response::builder().status(StatusCode::FORBIDDEN).body(Body::from("No auth header\n")),
+        Some(h) => h.to_str().unwrap(),
+    };
+    if expected_pass != provided_pass {
+        return Response::builder().status(StatusCode::FORBIDDEN).body(Body::empty());
+    }
+
+    let device = match env::var("DEVICE") {
+        Err(_) => return Response::builder().status(StatusCode::INTERNAL_SERVER_ERROR).body(Body::from("Misconfigured app\n")),
+        Ok(r) => r
+    };
+    let token = match env::var("TOKEN") {
+        Err(_) => return Response::builder().status(StatusCode::INTERNAL_SERVER_ERROR).body(Body::from("Misconfigured app\n")),
+        Ok(r) => r
+    };
+
+    let wanted_status = match get_device_status(&token, &device) {
+        Err(status) => return Response::builder().status(status).body(Body::empty()),
+        Ok(s) => match s.as_str() {
+            "off" => "on",
+            "on" => "off",
+            _ => return Response::builder().status(StatusCode::NOT_FOUND).body(Body::from("Misconfigured app\n")),
+        }
+    };
+
+    let res = match send_device_command(&token, &device, wanted_status) {
+        Err(status) => status,
+        Ok(s) => match s.as_str() {
+            "ACCEPTED" => StatusCode::NO_CONTENT,
+            _ => StatusCode::NOT_FOUND
+        }
+    };
+
+    Response::builder().status(res).body(Body::empty())
+}
+
+fn get_device_status(token: &str, device: &str) -> Result<String, StatusCode> {
+    let req = Request::builder()
+        .method(Method::GET)
+        .header(header::ACCEPT, "application/json")
+        .header(header::AUTHORIZATION, "Bearer ".to_string() + token)
+        .header(header::PRAGMA, "no-cache")
+        .uri(API_BASE.to_string() + device + "/status")
+        .body(Body::empty())
+        .expect("error building the request");
+
+    let rsp = match request(req) {
+        Err(status_code) => return Err(status_code),
+        Ok(r) => r,
+    };
+
+    let json: serde_json::Value = match serde_json::from_str(String::from_utf8(rsp.body().to_vec()).expect("getting device status").as_str()) {
```

Comment:
Is better to deserialise it to some struct of objects. In this case it will validate the schema and structure of json.

Reply:
I prefer to keep it simple, it is just an example

Reply:
We have handle all possible errors and to avoid panic which will be catch by fastedge runtime whitout much details. And as it is a public demo, it should be ideal guide for other.